### PR TITLE
Filter items with wildcard

### PR DESCRIPTION
**Pull request by ecomodeller**
Example with dfs0 but works with any dfs file
image

### CODE REVIEW

**Comment by jsmariegaard** [comment]
❤️

**Comment by jsmariegaard** [comment]
Would be great to add this to Dataset.__getitem__ https://github.com/DHI/mikeio/blob/72842c3d5d4daf1e66a06f3a9eeb17d743605511/mikeio/dataset.py#L727 too, for consistency

**Comment by ecomodeller** [comment]
> Would be great to add this to Dataset.**getitem**
> 
> https://github.com/DHI/mikeio/blob/72842c3d5d4daf1e66a06f3a9eeb17d743605511/mikeio/dataset.py#L727
> 
> too, for consistency

✔️

**Review by Hendrik1987 — APPROVED** [review]
Thanks @ecomodeller , great feature!

- I tested with .dfs2 and .dfsu point spectra, works as expected and with this my use case is covered!
- As cherry on the cake: Would there be an option / hard to implement to allow something like `mikeio.open(fn, items='Point 1*')` to check the effect before loading (big) data into memory. However, can also be tested with `mikeio.read(time=0, items='Point 1*')`, so this is really only nice to have.
- When testing I noticed that `ds.

File: mikeio/dfsutil.py [hunk 1]
```diff
@@ -20,7 +29,10 @@ def _valid_item_numbers(
         return list(range(n_items_file))
 
     if np.isscalar(items):
-        items = [items]
+        if isinstance(items, str) and "*" in items:
```
[the remark this answers is not in the document]
Comment (ecomodeller) [reply mikeio/dfsutil.py:32]: yes, np.isscalar doesn't modify the input
```python
>>> import numpy as np
>>> items = "foo"
>>> np.isscalar(items)
True
>>> items
'foo'
```

**Pull request merged**